optuna_dashboard/ts/components/TrialList.tsx

```diff
@@ -437,6 +445,130 @@ const TrialArtifact: FC<{ trial: Trial }> = ({ trial }) => {
                 </CardContent>
               </Card>
             )
+          } else if (
+            modelViewerTargetExt.includes(a.filename.split(".").pop() || "")
+          ) {
+            return (
+              <Card
+                key={a.artifact_id}
+                sx={{
+                  marginBottom: theme.spacing(2),
+                  display: "flex",
+                  flexDirection: "column",
+                  width: width,
+                  minHeight: "100%",
+                  margin: theme.spacing(0, 1, 1, 0),
+                }}
+              >
+                <Box
+                  sx={{
+                    flexGrow: 1,
+                    display: "flex",
+                    justifyContent: "center",
+                    alignItems: "center",
+                  }}
+                >
+                  <ModelViewer
+                    src={`/artifacts/${trial.study_id}/${trial.trial_id}/${a.artifact_id}`}
+                    width={width}
+                    height={height}
+                    hasGizmo={false}
+                    filetype={a.filename.split(".").pop()}
+                  />
+                </Box>
+                <CardContent
+                  sx={{
+                    display: "flex",
+                    flexDirection: "row",
+                    padding: `${theme.spacing(1)} !important`,
+                  }}
+                >
+                  <Typography
+                    sx={{
+                      p: theme.spacing(0.5, 0),
+                      flexGrow: 1,
+                      wordWrap: "break-word",
+                      maxWidth: `calc(100% - ${theme.spacing(12)})`,
+                    }}
+                  >
+                    {a.filename}
+                  </Typography>
+                  <IconButton
+                    aria-label="show artifact 3d model"
+                    size="small"
+                    color="inherit"
+                    sx={{ margin: "auto 0" }}
+                    onClick={() => {
+                      setOpen3dModelViewer(() => {
+                        const obj = { ...open3dModelViewer }
+                        obj[a.filename] = true
+                        return obj
+                      })
+                    }}
+                  >
+                    <FullscreenIcon />
+                  </IconButton>
+                  <Modal
+                    open={
+                      a.filename in open3dModelViewer
+                        ? open3dModelViewer[a.filename]
```

Comment:
We can upload multiple artifacts with the same filename. How about using `artifact_id` instead?

```suggestion
                        ? open3dModelViewer[a.artifact_id]
```